codestodians/food-basket-api
# [Feature: #154623848] Update the `last_login` field for any user.

#### What does this PR do?
- It makes available the data for when a user last logged in.

#### Description of Task to be completed?
- Update the `last_login` field in the Login mutation
- Update the `last_login` field immediately a user is created and logged in

#### How should this be manually tested?
- Check out into the `ft-implement-last-login-detail-154623848` branch.
- Call the `userLogin` as well as `create_user` mutations on `graphiql`.
- Check the DB to see that the `last_login` field is now updated.

#### Any background context you want to provide?
- I believe that populating this field will be useful in the long run especially on the frontend in cases when we want to make batch calls/mutations instead of single multiple mutations (that will result in slower load time). There'd be a check for when the user last logged in, and if for instance items to be added to the cart have stayed so long in the application state, a single call could be made to add them to cart on the backend.
- I also believe that this feature could be useful in other ways not immediately apparent now.

#### What are the relevant pivotal tracker stories?
- [#154623848](https://www.pivotaltracker.com/story/show/154623848)

#### Screenshots (if appropriate)

<img width="1280" alt="screen shot 2018-01-25 at 12 59 41 pm" src="https://user-images.githubusercontent.com/22825700/35387344-a6013004-01cf-11e8-8a5e-9a4b3790733d.png">
